core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -121,44 +121,44 @@ class AdminClient(val time: Time,
     listAllGroupsFlattened.filter(_.protocolType == ConsumerProtocol.PROTOCOL_TYPE)
   }
 
-  def describeGroup(groupId: String): GroupSummary = {
+  def getConsumerGroupSummary(groupId: String): ConsumerGroupSummary = {
     val coordinator = findCoordinator(groupId)
     val responseBody = send(coordinator, ApiKeys.DESCRIBE_GROUPS, new DescribeGroupsRequest(List(groupId).asJava))
     val response = new DescribeGroupsResponse(responseBody)
     val metadata = response.groups().get(groupId)
     if (metadata == null)
       throw new KafkaException(s"Response from broker contained no metadata for group $groupId")
+    if (metadata.state != "Dead" && metadata.protocolType() != ConsumerProtocol.PROTOCOL_TYPE)
+      throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType()}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
     val members = metadata.members().map { member =>
       val metadata = Utils.readBytes(member.memberMetadata())
       val assignment = Utils.readBytes(member.memberAssignment())
       MemberSummary(member.memberId(), member.clientId(), member.clientHost(), metadata, assignment)
     }.toList
-    GroupSummary(metadata.state(), metadata.protocolType(), metadata.protocol(), members)
+    ConsumerGroupSummary(metadata.state(), metadata.protocol(), members, coordinator)
   }
 
   case class ConsumerSummary(memberId: String,
                              clientId: String,
                              clientHost: String,
                              assignment: List[TopicPartition])
 
-  def describeConsumerGroup(groupId: String): Option[List[ConsumerSummary]] = {
-    val group = describeGroup(groupId)
-    if (group.state == "Dead")
-      return None
-
-    if (group.protocolType != ConsumerProtocol.PROTOCOL_TYPE)
-      throw new IllegalArgumentException(s"Group $groupId with protocol type '${group.protocolType}' is not a valid consumer group")
-
-    if (group.state == "Stable") {
-      Some(group.members.map { member =>
-        val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(member.assignment))
-        new ConsumerSummary(member.memberId, member.clientId, member.clientHost, assignment.partitions().asScala.toList)
-      })
-    } else {
-      Some(List.empty)
-    }
+  def describeConsumerGroup(groupId: String): (String, Option[(Node, List[ConsumerSummary])]) = {
```

Comment:
Currently `ConsumerGroupSummary` has a field for the group members, which are represented as instances of `MemberSummary`. I'm wondering if it would make sense to use `ConsumerSummary` instead. Then we might only need a single method returning `ConsumerGroupSummary`.